src/ledger/pathfind.ts

```diff
@@ -93,11 +95,13 @@ function filterSourceFundsLowPaths(pathfind: PathFind,
 ): RippledPathsResponse {
   if (pathfind.source.amount &&
       pathfind.destination.amount.value === undefined && paths.alternatives) {
-    paths.alternatives = _.filter(paths.alternatives, alt => {
-      return alt.source_amount &&
-        pathfind.source.amount &&
+    paths.alternatives = _.filter(paths.alternatives, alt =>
+        !!alt.source_amount &&
+        !!pathfind.source.amount &&
+        // TODO: Returns false when alt.source_amount is a string. Fix?
```

Comment:
How can `alt.source_amount` be a string?

Reply:
` alt.source_amount` is of type `RippledAmount`, which can be a string. https://ripple.com/build/rippled-apis/ seems to document a few places where source_amount is a string.

Reply:
Looks like `source_amount` is only a string when the currency is XRP, in which case there is no need for pathfinding. So I think it's fine to return false in that case.

Reply:
>>> Looks like source_amount is only a string when the currency is XRP, in which case there is no need for pathfinding.

You can send XRP and use paths to send the recipient USD ... 

What you mean no need for path finding ?